### PR TITLE
Extend frequency capping to all dismissible prompts

### PR DESCRIPTION
**Pull request by justinchou-google**
b/303458595, b/303459208 Frequency capping is applied to `:basic` premonetization and `:openaccess` subscription posts. This covers ALL dismissible prompts.

### CODE REVIEW

File: src/runtime/auto-prompt-manager.ts [hunk 1]
```diff
@@ -690,12 +699,15 @@ export class AutoPromptManager {
...
+    }
+
     // FrequencyCapConfig may be undefined ONLY for subscription openacess
     // content. If so, display first valid action.
-    if (
-      this.isSubscription_(autoPromptType) ||
-      !this.isValidFrequencyCap_(frequencyCapConfig)
-    ) {
+    const isLockedContent =
+      this.isSubscription_(autoPromptType) && this.pageConfig_.isLocked();
```
Comment (oyj9109) [inline src/runtime/auto-prompt-manager.ts:709]: Would `isClosable` be more accurate than subscription && locked?
What's the expected behavior for non-subscription && (isClosable == false)?

File: src/runtime/auto-prompt-manager.ts [hunk 2]
```diff
@@ -704,8 +716,11 @@ export class AutoPromptManager {
       return actions[0];
     }
 
-    const globalFrequencyCapDuration =
-      frequencyCapConfig?.globalFrequencyCap?.frequencyCapDuration;
+    const isSubscriptionOpenaccess =
+      this.isSubscription_(autoPromptType) && !this.pageConfig_.isLocked();
+    const globalFrequencyCapDuration = isSubscriptionOpenaccess
+      ? SUBSCRIPTION_OPENACCESS_GLOBAL_FREQUENCY_CAP_DURATION
+      : frequencyCapConfig?.globalFrequencyCap?.frequencyCapDuration;
```
Comment (oyj9109) [inline src/runtime/auto-prompt-manager.ts:723]: Can we set up default value for global frequency in the server? Then I guess subscription & openaccess would run through the rest of logic.

Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:723]: Will move this to server side.

File: src/runtime/auto-prompt-manager.ts (hunk 1 above)
Comment (justinchou-google) [reply src/runtime/auto-prompt-manager.ts:709]: Discussed offline, extending to `isClosable` prompts.

**Review by oyj9109 — APPROVED** [review]

**Pull request merged**